hazelcast-client/src/test/java/com/hazelcast/client/usercodedeployment/ClientUserCodeDeploymentConfigTest.java

```diff
@@ -136,4 +136,21 @@ private void assertClassLoaded(List<Map.Entry<String, byte[]>> list, String name
         fail();
     }
 
+    @Test
+    public void testConfigWithJarFile_withInnerAndAnonymousClass() throws IOException, URISyntaxException, ClassNotFoundException {
+        ClientUserCodeDeploymentConfig config = new ClientUserCodeDeploymentConfig();
+        config.setEnabled(true);
+        ClassLoader classLoader = getClass().getClassLoader();
+        URL resource = classLoader.getResource("EntryProcessorWithAnonymousAndInner.jar");
```

Comment:
It is already there. :)
https://github.com/hazelcast/hazelcast/pull/10788/files#diff-fb35a9f4a7057ffecdf06164fe790e96